test/common/common/utility_test.cc

```diff
@@ -116,6 +117,54 @@ TEST(DateUtil, NowToMilliseconds) {
   EXPECT_EQ(12345067, DateUtil::nowToMilliseconds(test_time));
 }
 
+TEST(OutputBufferStream, All) {
+  // Fails on write to empty buffer
+  {
+    std::string data = "123";
+    OutputBufferStream ostream{nullptr, 0};
+    ASSERT_TRUE(ostream.good());
+
+    ostream << data;
+
+    EXPECT_TRUE(ostream.bad());
+  }
+
+  // Can write to buffer
+  {
+    std::string data = "123";
+    std::string buffer(3, 'x');
```

Comment:
This and the others could be `std::array`; that saves you a little on runtime since the contents of the buffer before streaming don't actually matter.

Reply:
I don't understand the `std::array` issue. I was suggesting declaring `std::array<uint8, 3> buffer` and using `std::array::data()` and `size()` below to access the raw bytes.

Reply:
Now use `std::array<char, N> ` for the buffer, thanks for clarifying